web-console/3-project-details/labels.md

```diff
@@ -0,0 +1,104 @@
+# Labels on Detail Pages
+
+## View Labels
+- Labels are currently displayed at the top of most detail pages (Deployments, Pods, Builds, Images, Storage, etc)
+- A new `Labels` section should be added to the bottom of the `Details` tab on all of these pages to allow for a more comprehensive labels area
+
+![Label 1](img/new-labels-00-view.png)
+
+## Add Labels
+![Label 1](img/new-labels-01-add.png)
+
+- A `Manage Labels` link should be displayed below the applied labels
+- Clicking the link will open a modal, displaying the labels that are already applied, as well as an option to add new labels.
+- Clicking `Add` will display a new row with two empty text fields for label Name and label Value.
+
+![Label 1](img/new-labels-02-add.png)
+
+- The save button should not be enabled until valid entries are made.
+
+![Label 1](img/new-labels-03-add.png)
+
+- Invalid characters will result in a field level error once the user changes focus away from the particular field.
+
+![Label 1](img/new-labels-03B-add.png)
+
+- Label names cannot have more than one value associated.
+- Attempting to enter a label name that has already been applied with a value will result in a field level error as well.
+
+![Label 1](img/new-labels-04-add.png)
+
+![Label 1](img/new-labels-05-add.png)
+
+- Once the user clicks `Save` the new label should appear at the bottom of the list of labels on the Details tab.
+
+## Edit Labels
+![Label 1](img/new-labels-01-edit.png)
+
+- The same `Manage Labels` link will be used for editing existing labels.
+- Users can change a name or value as desired, as long as the name is not already associated with another value.
```

Comment:
Will entering a name that already exists for the item cause a field-level error? Or some other kind of error?

Reply:
I believe I showed an example of this @ncameronbritt where the name "app" already exists and a user tried to add another "app" label. Let me know if you have questions still!

Reply:
Ah, ok I see. I wonder if we can do better then telling them only that it's invalid. If a label is already used, could we say "This label already exists" or something? And in the case of illegal characters we could say something like "Names can only contain alphanumeric characters" (or whatever the rules happen to be.)

Reply:
Agreed, I'll update these!